stable/redis/templates/redis-slave-deployment.yaml

```diff
@@ -0,0 +1,114 @@
+      {{ include "redis.slave.securityContext" . | indent 6 }}  
+      {{- if (.Values.slave.nodeSelector | default .Values.nodeSelector) }}
+      nodeSelector:
+{{ toYaml (.Values.slave.nodeSelector | default .Values.nodeSelector) | indent 8 }}
+      {{- end }}
+      {{- if (.Values.slave.tolerations | default .Values.tolerations) }}
+      tolerations:
+{{ toYaml (.Values.slave.tolerations | default .Values.tolerations) | indent 8 }}
+      {{- end }}
+      containers:
+      - name: {{ template "redis.fullname" . }}
+        image: {{ template "redis.image" . }}
+        imagePullPolicy: {{ .Values.imagePullPolicy | default "" | quote }}
+        {{- if (.Values.slave.args | default .Values.master.args) }}
+        args:
+{{ toYaml (.Values.slave.args | default .Values.master.args) | indent 10 }}
+        {{- end }}
+        env:
+        - name: REDIS_REPLICATION_MODE
+          value: slave
+        - name: REDIS_MASTER_HOST
+          value: {{ template "redis.fullname" . }}-master
+        - name: REDIS_PORT
+          value: {{ .Values.slave.port | default .Values.master.port | quote }}
+        - name: REDIS_MASTER_PORT_NUMBER
+          value: {{ .Values.master.port | quote }}
+        {{- if .Values.slave.usePassword }}
+        - name: REDIS_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              {{- if .Values.master.existingSecret }}
+              name: {{ .Values.master.existingSecret }}
+              {{- else }}
+              name: {{ template "redis.fullname" . }}
+              {{- end }}
+              key: redis-password
+        - name: REDIS_MASTER_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              {{- if .Values.master.existingSecret }}
+              name: {{ .Values.master.existingSecret }}
+              {{- else }}
+              name: {{ template "redis.fullname" . }}
+              {{- end }}
+              key: redis-password
+        {{- else }}
+        - name: ALLOW_EMPTY_PASSWORD
+          value: "yes"
+        {{- end }}
+        - name: REDIS_DISABLE_COMMANDS
+          value: {{ .Values.slave.disableCommands | default .Values.master.disableCommands }}
+        {{- if (.Values.slave.extraFlags | default .Values.master.extraFlags) }}
+        - name: REDIS_EXTRA_FLAGS
+          value: {{ .Values.slave.extraFlags | default .Values.master.extraFlags | join " " }}
+        {{- end }}
+        ports:
+        - name: redis
+          containerPort: {{ .Values.slave.port | default .Values.master.port }}
+        {{- /* Include master livenessProbe if slave livenessProbe not defined */ -}}
+        {{ include "redis.slave.livenessProbe" . | indent 8 }}  
+        {{- /* Include master readinessProbe if slave readinessProbe not defined */ -}}
+        {{ include "redis.slave.readinessProbe" . | indent 8 }}  
+        resources:
+{{ toYaml (.Values.slave.resources | default .Values.resources) | indent 10 }}
+  {{- if .Values.metrics.enabled }}
```

Comment:
do we need to have a metrics exporter for every container? If the exporter exposes metrics of all the nodes in the cluster then we can just have one exporter in a different deployment object.

Reply:
My concern is the case where the read-only slave doesn't have authentication enabled and the master has. In this case we would need to create like two metrics deployments, one for the slaves and one for the master. Would it make sense to unify the usePassword authentication part?

Reply:
I find it hard to imagine a use case where only the master has authentication enabled. IMO authentication would be enabled on the entire cluster, so yes to me it makes sense to unify the `usePassword` as a global configuration. wdyt @tomas?

Reply:
agree! I can't imagine a use case where master has auth and slaves does not

Reply:
@sameersbn @tompizmor unified